pkg/furnace/furnace.go

```diff
@@ -0,0 +1,67 @@
+package furnace
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"google.golang.org/grpc"
+
+	api "github.com/prmsrswt/foundry/pkg/furnace/furnacepb"
+)
+
+type Furnace struct {
+	mx sync.RWMutex
+	// TODO(prmsrswt): Implement better queueing system. Something like https://github.com/sheerun/queue.
+	buildQueue     []Package
```

Comment:
the build queue was supposed to be in store room if I remember 
and furnace was to just store its status that what its building and receive build commands from the store room.
Instead of maintaining its own queue